src/components/CheckboxField/__tests__/CheckboxField.spec.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import {
+  expect,
+  test,
+} from '@playwright/experimental-ct-react';
+import {
+  mixPropTests,
+  propTests,
+} from '../../../../tests/playwright';
+import {
+  CheckboxFieldForTest,
+  CheckboxFieldForRefTest,
+  CheckboxWForFormLayoutTests,
+} from './CheckboxField.story';
+import { checkedPropTest } from './_propTests/checkedPropTest';
+
+test.describe('CheckboxField', () => {
+  test.describe('base', () => {
+    test.describe('visual', () => {
+      [
+        ...propTests.labelPropTest,
+        ...propTests.renderAsRequiredPropTest,
+        ...propTests.isLabelVisiblePropTest,
+        ...propTests.labelPositionPropTest(),
+        ...mixPropTests([
+          checkedPropTest,
+          propTests.disabledPropTest,
+          propTests.validationStatePropTest(true),
+        ]),
+        ...mixPropTests([
+          checkedPropTest,
+          propTests.requiredPropTest,
+        ]),
+        ...mixPropTests([
+          checkedPropTest,
+          propTests.validationTextPropTest(true),
+          propTests.validationStatePropTest(true),
+        ]),
+        ...mixPropTests([
+          checkedPropTest,
+          propTests.validationTextPropTest(true),
+          propTests.helpTextPropTest(true),
+        ]),
+      ].forEach(({
+        name,
+        onBeforeTest,
+        props,
+      }) => {
+        test(name, async ({
+          mount,
+          page,
+        }) => {
+          if (onBeforeTest) {
+            await onBeforeTest(page);
+          }
+
+          const component = await mount(
+            <CheckboxFieldForTest
+              {...props}
+            />,
+          );
+
+          const screenshot = await component.screenshot();
+          expect(screenshot).toMatchSnapshot();
+        });
+      });
+    });
+
+    test.describe('non-visual', () => {
+      test('id', async ({ mount }) => {
+        const idValue = 'checkbox-id';
+        const label = 'checkbox-label';
+        const helpText = 'checkbox-helpText';
+        const validationText = 'checkbox-validationText';
+
+        const component = await mount(
+          <CheckboxFieldForTest
+            helpText={helpText}
+            id={idValue}
+            label={label}
+            validationText={validationText}
+          />,
+        );
+
+        await expect(component.getByRole('checkbox')).toHaveAttribute('id', idValue);
```

Comment:
This is quite interesting! But as I know RUI, there are components for those such approach will not be enough. lets discuss it